turbo/txpool-provider/control.go

```diff
@@ -0,0 +1,168 @@
+package txpoolprovider
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	"github.com/ledgerwatch/turbo-geth/common"
+	"github.com/ledgerwatch/turbo-geth/core/rawdb"
+	"github.com/ledgerwatch/turbo-geth/core/types"
+	"github.com/ledgerwatch/turbo-geth/core/types/accounts"
+	"github.com/ledgerwatch/turbo-geth/ethdb"
+	"github.com/ledgerwatch/turbo-geth/ethdb/remote/remotedbserver"
+	"github.com/ledgerwatch/turbo-geth/params"
+	"github.com/ledgerwatch/turbo-geth/turbo/adapter"
+	"github.com/ledgerwatch/turbo-geth/turbo/txpool-provider/pb"
+)
+
+type TxPoolControlServer struct {
+	pb.UnimplementedTxpoolControlServer
+	chainConfig *params.ChainConfig
+	kv          ethdb.KV
+	events      *remotedbserver.Events
+}
+
+func NewTxPoolControlServer(cc *params.ChainConfig, kv ethdb.KV, events *remotedbserver.Events) *TxPoolControlServer {
+	return &TxPoolControlServer{chainConfig: cc, kv: kv, events: events}
+}
+
+func (c *TxPoolControlServer) AccountInfo(ctx context.Context, request *pb.AccountInfoRequest) (*pb.AccountInfoReply, error) {
+	var (
+		db   = ethdb.NewObjectDatabase(c.kv)
+		hash = common.BytesToHash(request.BlockHash)
+		addr = common.BytesToAddress(request.Account)
+	)
+
+	head, err := rawdb.ReadHeaderByHash(db, hash)
+	if err != nil {
+		return nil, err
+	}
+
+	acc, err := readAccountAtBlock(ctx, c.kv, addr, head.Number.Uint64())
```

Comment:
Can you use `rawdb.PlainReadAccount` instead of that? You need the latest view of the account, right?

Reply:
The interface for this is actually read an account at a [certain block hash](https://github.com/ledgerwatch/interfaces/blob/3e877092f8349d55f0f688df6f5560e1dcb6419c/txpool/txpool_control.proto#L8). Is there a better way to do this?